src/CartBlock.C

```diff
@@ -232,27 +222,27 @@ void CartBlock::insertInInterpList(int procid,int remoteid,int remoteblockid,dou
 void CartBlock::insertInDonorList(int senderid,int index,int meshtagdonor,int remoteid,int remoteblockid, double cellRes)
 {
   DONORLIST *temp1;
-  int ijklmn[6];
+  int i,j,k,x_stride,xy_stride;
   int pointid;
   temp1=(DONORLIST *)malloc(sizeof(DONORLIST));
-  amr_index_to_ijklmn(dims[0],dims[1],dims[2],nf,index,ijklmn);
-  //pointid=ijklmn[5]*(pdegree+1)*(pdegree+1)*d3+
-  //        ijklmn[4]*(pdegree+1)*d3+
-  //        ijklmn[3]*d3+
-  //        ijklmn[2]*d2+
-  //        ijklmn[1]*d1+
-  //        ijklmn[0];
-  pointid=(ijklmn[2]*d2+ijklmn[1]*d1+ijklmn[0])+
-           ijklmn[5]+
-           ijklmn[4]+ijklmn[3];
+
+  // Get point-id accounting for nf
+  x_stride = (dims[0]+2*nf);
+  xy_stride = x_stride*(dims[1]+2*nf);
+  index %= d3nf;
```

Comment:
What is the need for this line ?

Reply:
@jsitaraman Thanks for catching this! It was unnecessary, since `index < d3nf ` after removal of `p>0` I have fixed this in a commit.